### PR TITLE
Uploading metrics to gsheet for rename tests

### PR DESCRIPTION
**Pull request by anushka567**
### Description
The [_get_values_to_export](https://github.com/GoogleCloudPlatform/gcsfuse/blob/d55798c89be1660c4f43a3c3c4bedb8657ca0844/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py#L67)  function takes in metrics dictionary and converts it into rows of values ready to be uploaded to google spreadsheet using [_upload_values_to_gsheet](https://github.com/GoogleCloudPlatform/gcsfuse/blob/d55798c89be1660c4f43a3c3c4bedb8657ca0844/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py#L46) 
### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Checked the entire script is working as intended and metrics are getting published in [google sheet](https://docs.google.com/spreadsheets/d/1UVEvsf49eaDJdTGLQU1rlNTIAxg8PZoNQCy_GX6Nw-A/edit?usp=sharing&resourcekey=0-crHvwEsEQbDD8xV1fy57-A)
2. Unit tests -  [checked correct format of data being generated from metrics dictionary for upload](https://github.com/GoogleCloudPlatform/gcsfuse/blob/d55798c89be1660c4f43a3c3c4bedb8657ca0844/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark_test.py#L169) , [checked correct error handling when gsheet id not passed](https://github.com/GoogleCloudPlatform/gcsfuse/blob/d55798c89be1660c4f43a3c3c4bedb8657ca0844/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark_test.py#L208)
3. Integration tests - NA

### CODE REVIEW

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py [hunk 1]
```diff
@@ -39,6 +43,76 @@
...
+  Writes rename results to Google Spreadsheets.
+  Args:
+    worksheet (str): Google sheet name to which results will be uploaded.
+    data (list): metrics to be uploaded.
+    spreadsheet_id: Google spreadsheet id.
+  """
+  # Changing directory to comply with "cred.json" path in "gsheet.py".
+  os.chdir('..')
+  exit_code = 0
+  if spreadsheet_id == "":
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py:57]: Will it be the case? Aren't we confirming while running python file only?

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py [hunk 2]
```diff
@@ -271,3 +345,14 @@ def _parse_arguments(argv):
...
   flat_parsed_metrics = _parse_results(dir_str, results['flat'], args.num_samples)
+  upload_values_flat = _get_values_to_export(dir_str, flat_parsed_metrics,
+                                             "flat")
+
+  if args.upload_gs:
+    log.info('Uploading files to the Google Sheet\n')
+    exit_code = _upload_to_gsheet(WORKSHEET_NAME_FLAT, upload_values_flat,
+                                  SPREADSHEET_ID)
+    if exit_code != 0:
+      log.error("Upload to gsheet unsuccessful!")
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py:356]: unsuccessful --> fail

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py [hunk 3]
```diff
@@ -39,6 +43,76 @@
 log = logging.getLogger()
 
 
+def _upload_to_gsheet(worksheet, data, spreadsheet_id) -> (int):
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py:46]: I think we have this function on listing_benchmark also. Can we move that in to util?

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py (hunk 1 above)
Comment (anushka567) [reply perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py:57]: spreadsheet_id is optional argument ,hence , required.

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py (hunk 3 above)
Comment (anushka567) [reply perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py:46]: Due to difference in function behavior, making this function private to rename benchmark.

**Review by Tulsishah — APPROVED** [review]

**Pull request merged**